bond/bond.py

```diff
@@ -178,19 +178,22 @@ def change_key_groups(self, og_prefix, new_prefix):
                     # change each filename according to new key group
                     self.change_filename(file_path, new_entities)
 
-        with open(new_prefix + '_change_files.sh', 'w') \
-                as exe_script:
-            for old, new in zip(self.old_filenames, self.new_filenames):
-                exe_script.write("mv %s %s\n" % (old, new))
+            self.datalad_save()
 
-        my_proc = subprocess.run(
-            ['bash', new_prefix + '_change_files.sh'])
+            # create string of mv command ; mv command for dlapi.run
+            mv_str = ''
```

Comment:
```
move_ops = []
for from_file, to_file in zip(self.old_filenames, self.new_filenames):
    move_ops.append('mv %s %s' % (from_file, to_file)
move_cmd = ': '.join(move_ops)
```

Please try to avoid [i]'s in loops